### ISSUE TITLE
[Image] Make ALAsset representation and orientation configurable

### ISSUE THREAD

**Issue by VonD** [facebook/react-native#1567]
The Image component currently always renders the full resolution image, which can be particularly problematic when loading several images from the assets library.

This issue is a follow up from #930

**Comment by asamiller** [comment]
+1

**Comment by bparadie** [comment]
@VonD @brentvatne I believe https://github.com/facebook/react-native/pull/1406 addresses this issue. Am I right?

**Comment by brentvatne** [comment]
@bparadie - seems to fix the size portion at least, I'm not familiar with this too much - is there a native api for configuring the orientation as well?

**Comment by bparadie** [comment]
@brentvatne No, unfortunately #1406 only addresses the resolution part and allows requesting thumbnails. If it was up to me I would split this issue up into two separate issues and leave the request for having the orientation configurable open. Cheers.

**Comment by brentvatne** [comment]
@bparadie - Makes sense. What's a use case for configuring the orientation? How does that work? Aren't thumbnails square?

**Comment by bparadie** [comment]
@brentvatne I actually care less for the orientation case. My thumbnails are squares. But I can see @VonD's point. The much bigger problem was performance. Pulling in full res images killed the performance if you want to display a grid with tiles showing what's on your camera roll. 

With @Iragne's PR things got so fast that I am tempted to take out the code that synchronously queues up the images and let them flow in - fire hose style. I think the iPhone can handle async processing of thumbnails.

I am referring to this comment/code in `ImageLoader.m`:
```objective-c
        // ALAssetLibrary API is async and will be multi-threaded. Loading a few full
        // resolution images at once will spike the memory up to store the image data,
        // and might trigger memory warnings and/or OOM crashes.
        // To improve this, process the loaded asset in a serial queue.
```

I wonder whether that serial queue is really necessary if `thumbs` is `YES` ...

**Comment by bparadie** [comment]
Yep, the thumbnail case does not need a serial queue - my iPhone 6 could digest over 200 videos from the cameraroll without any problems. I've provided the code snippet at https://github.com/facebook/react-native/pull/1406#issuecomment-113709903 .

**Comment by VonD** [comment]
@bparadie @brentvatne the reason I included the orientation within the same issue is that when i first tried replacing `[representation fullResolutionImage]` by `[asset thumbnail]` in the ImageLoader, I ended up with some thumbnails upside-down (e.g. if the picture was taken with the home button on top), other rotated -90° etc.. which didn't seem to happen with the `fullResolutionImage`.


I didn't dig into this too deeply, but the following snippet solved my issue : 
````
UIImage *image = [UIImage imageWithCGImage:[asset thumbnail] scale:1.0f orientation:UIImageOrientationUp];
RCTDispatchCallbackOnMainQueue(callback, nil, image);
````

**Comment by myusuf3** [comment]
@VonD how do we go about fixing this orientation issue? The changes you made edit the react objc code?

**Comment by VonD** [comment]
@myusuf3 for the moment you'll have to patch this : https://github.com/facebook/react-native/blob/master/Libraries/Image/RCTImageLoader.m#L78

**Comment by myusuf3** [comment]
@VonD is there a fix in 0.6.0?

**Comment by aroth** [comment]
> If orientation really does have a different behavior depending on representation, then maybe it's best to set it explicitly

@myusuf3 @VonD [asset thumbnail] can only have an orientation of ALAssetOrientationUp.  Alternatively, fullResolution or fullScreen representations can use any available orientation. I have code nearly ready that allows passing in both orientation and representation values to an <Image ...> tag.

**Comment by aroth** [comment]
@brentvatne   These issues have been resolved as of 0.9 and 0.11
